doc/specs/stdlib_hash_procedures.md

```diff
@@ -0,0 +1,1748 @@
+
+##### Class
+
+Subroutine
+
+##### Argument
+
+`harvest`: Shall be an integer of kind `INT64`. It is an `intent(out)`
+argument.
+
+##### Note
+
+`ODD_RANDOM_INTEGER` is intended to generate seeds for
+ `UNIVERSAL_MULT_HASH`. `ODD_RANDOM_NUMBER` uses Fortran's intrinsic
+ `RANDOM_NUMBER` and the values returned can be changed by calling the
+ intrinsic `RANDOM_INIT`.
+
+##### Example
+
+See `UNIVERSAL_MULT_HASH`.
+
+
+#### `PENGY_HASH` - maps a character string or integer vector to an integer
+
+##### Status
+
+Experimental
+
+##### Description
+
+Maps a character string or integer vector to a 64 bit integer whose
+value also depends on a scalar 32 bit integer, `seed`.
+
+##### Syntax
+
+`code = [[stdlib_64_bit_hash_codes:pengy_hash]]( key, seed )`
+
+#####  Class
+
+Pure/elemental function
+
+##### Arguments
+
+`key`: shall be a scalar  expression of type default character or a
+Rank 1 integer vector expression of kind `INt8`, `INT16`, `INT32`, or
+`INTT64`. It is an `intent(in)` argument.
+
+`seed`: shall be an integer expression of kind `INT64`. It is
+an `intent(in)` argument.
+
+##### Result
+
+The result is an integer of kind `INT64`.
+
+##### Note
+
+`PENGY_HASH` is an implementation of the 64 bit `pengyhash` of Alberto
+Fajardo. The hash has acceptable performance on small keys, and good
+performance on long keys. It passes all the SMHasher tests, and has
+no known bad seeds.
+It is a *pure* function for integer arrays, and an *elemental*
+function for character strings. 
+
+##### Exampl
+
+```fortran
+    program demo_pengy_hash
+      use stdlib_64_bit_hash_codes, only: new_pengy_hash_seed, pengy_hash
+      use iso_fortran_env, only: int32, int64 
+      implicit none
+      integer, allocatable :: key(:)
+      integer(int64) :: hash
+      integer(int32)  ::  seed
+      key = [ 0, 1, 2, 3 ]
+      seed = 0_int32
+      call new_pengy_hash_seed( seed )
+      hash = pengy_hash( key, seed )
+      print *, seed, hash
+    end program demo_pengy_hash
+```
+
+
+#### `SPOOKY_HASH` - maps a character string or integer vector to an integer
+
+##### Status
+
+Experimental
+
+##### Description
+
+Maps a character string or integer vector to a 64 bit integer whose
+value also depends on a two element vector,  `seed`.
+
+##### Syntax
+
+`code = [[stdlib_64_bit_hash_codes:spooky_hash]]( key, seed )`
+
+#####  Class
+
+Function
+
+##### Arguments
+
+`key`: shall be a scalar of type default character expression or a
+Rank 1 integer vector expression of kind `INt8`, `INT16`, `INT32`, or
+`INTT64`. It is an `intent(in)` argument.
+
+`seed`: shall be a two element integer vector expression of kind
+`INT64`. It is an `intent(in)` argument.
+
+##### Result
+
+The result is a two element integer vector of kind `INT64`.
+
+##### Note
+
+`SPOOKY_HASH` is an implementation of the 64 bit version 2 of
+SpookyHash of Bob Jenkins. The code was designed for Little-Endian
+compilers. The output is different on Big Endian compilers, but still
```

Comment:
```suggestion
SpookyHash of Bob Jenkins. The code was designed for little-endian
compilers. The output is different on big-endian compilers, but still
```
I think uppercase initial are unwarranted here. If emphasis is needed I would use italics.

Reply:
Capitalization is unwarranted and a hyphen is required. Done.